### ISSUE TITLE
Tower-cli config support in python

### ISSUE THREAD

**Issue by dariaknl** [ansible/tower-cli#149]
There is a python module 'tower-cli' available which allows to use tower-cli api in the scripts. But you can only make it work if you have tower-cli.cfg created locally with _username, password, host, verify_ssl._
It would be a great to make it possible to set tower-cli configuration using this library somewhere as a part of:
`import tower_cli`
`tower_cli.conf.Settings`
so it is not needed to create either tower-cli.cfg or enter configuration manually from the command line.
(I am creating it after tracing this issue with ansible support, just in case :) )

**Comment by AlanCoding** [comment]
I would like to get a better picture of the use story that is desired. Let's say that I am user `admin` and I want to create a new user inside of a python script. The following script should work, while also avoiding editing of the config file.

```python
user_res = tower_cli.get_resource('user')

resp_json = user_res.create(
    username='new-user',
    password='pa$$word',
    email='email@email.com',
    tower_host="http://towerhost.company.com/",
    tower_username="admin",
    tower_password="pa$$word")
```

After this, you should be able to get the new user id from `resp_json['id']` if the creation was successful. Are there any problems or bugs with using this method?

Is the idea that you would like to be able to edit tower-cli.cfg from the python script? Or is it that you would like to edit the config inside the script without saving to the config file?

Thanks for giving us feedback.

**Comment by dariaknl** [comment]
- I tried the way you propose it when no tower-cli.cfg exists locally:


```
import tower_cli
def test_rest_get():
    user_res = tower_cli.get_resource('user')
    resp_json = user_res.create(
    username='library-demo',
    password='library9876',
    email='email@email.com',
    tower_host="https://example.com/",
    tower_username="NormalUser",
    tower_password="test")
```
It fails with:

`'Could not establish a secure connection. '
ConnectionError: Could not establish a secure connection. Please add the server to your certificate authority.
You can run this command without verifying SSL with the --insecure flag, or permanently disable verification by the config setting:
tower-cli config verify_ssl false
`


- Coming back to the initial use case. 
I want to be able to use tower_cli.api package
```
from tower_cli import api
def test_rest_get():
    c = api.Client()
    response = c.get('projects')
    print response.json()['count']
```
This will work only if I have locally tower-cli.cfg.
I want to get this working without creating locally tower-cli.cfg (and I though that should be possible so you can fully automate usage of your python scripts).

By the way I also tried to create tower-cli.cfg using python subprocess. It did not work well, the configuration is not updated immediately. If you are interested in these results more, I can give more info. 

To sum up:

- is it possible to use tower-cli api in python without creating locally tower-cli.cfg?
- if not, what is the proposed way to generate tower-cli.cfg within python?  
[The use case would be: I want to use tower-cli in python to script tests, which are checking some test data for different users - admin. user, superuser. So it means I need to change tower-cli configuration for every test]


Thanks!

**Comment by AlanCoding** [comment]
Got it. Thanks for the info, this should be enough to move on. So basically, we need a way to continue to execute a command without making a reference to the config file (or to manage a version cached in memory, like what was suggested in the original issue comment). I think this is a very reasonable expectation when using it as an import, which we do want to support.

**Comment by hufman** [comment]
My current approach is to do something like this at the beginning of my program:

```
    tower_cli.conf.settings._runtime.set("general", "host", host)
    tower_cli.conf.settings._runtime.set("general", "username", auth[0])
    tower_cli.conf.settings._runtime.set("general", "password", auth[1])
    tower_cli.conf.settings._cache.clear()
```

**Comment by defionscode** [comment]
you can also do it like so 
```python
from tower_cli import get_resource
from tower_cli import conf

with conf.settings.runtime_values(host='https://mytower/', username='jdavila', password='NSAp00f', verify_ssl=True):
    usr_r = get_resource('user')
    print usr_r.list()

**Comment by AlanCoding** [comment]
Let me link this Ansible module PR here:

https://github.com/ansible/ansible-modules-extras/pull/2810

Importing tower-cli as a dependency for this was under consideration. I think it would be good to resolve this issue first, to make the python usage a little bit more clear and more stable.

**Comment by AlanCoding** [comment]
Also, believed to be solved by #266

**Comment by jangsutsr** [comment]
Issue solved. Patch will ship with 3.1.1 release.
